storage/remote/client.go

```diff
@@ -198,11 +204,33 @@ func (c *Client) Store(ctx context.Context, req []byte) error {
 		err = errors.Errorf("server returned HTTP status %s: %s", httpResp.Status, line)
 	}
 	if httpResp.StatusCode/100 == 5 {
-		return RecoverableError{err}
+		return RecoverableError{err, defaultBackoff}
+	}
+	if httpResp.StatusCode == 429 {
```

Comment:
```suggestion
	if httpResp.StatusCode == http.StatusTooManyRequests {
```

Reply:
If we want to name via constants, then should other codes be changes similarly?